cogs/birthday/birthday.py

```diff
@@ -433,12 +434,42 @@ async def getSelfBirthday(self, ctx: Context):
                 if month and day:
                     birthday = date(2020, month, day)
                     birthdayStr = "{0:%B} {0:%d}".format(birthday)
-                    await ctx.author.send(
-                        f"{headerGood}: Your birthday is "
-                        f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
-                    )
-                    return
-
+                    try:
+                        await ctx.author.send(
+                            f"{headerGood}: Your birthday is "
+                            f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
+                        )
+                        return
+                    except discord.Forbidden:
+
+                        def check(msg: discord.Message):
+                            return msg.author == ctx.author and msg.channel == ctx.channel
+
+                        await ctx.send(
+                            f"{headerWarn}: I would like to DM you your birthday but it seeems that "
+                            "you have disabled DMs from this server. Would you still like to continue here? "
+                            "Your birthday will be sent here and deleted after a short delay. "
+                            f"\nType {bold('`yes`', escape_formatting=False)} to confirm. "
+                            "\nAnything else will be treated as no."
```

Comment:
No need trailing spaces when the following lines begin with newlines (`\n`).
Also I think the source code looks better when we don't blend in "\n", and the _[a]nything else will be treated as no_ can stay on the same line as the call-to-action line. Lemme know what you think.
```suggestion
                            "\n".join((
                                f"{headerWarn}: I would like to DM you your birthday but it seeems that "
                                "you have disabled DMs from this server. Would you still like to continue here? "
                                "Your birthday will be sent here and deleted after a short delay.",
                                f"Type {bold('`yes`', escape_formatting=False)} to confirm. "
                                "Anything else will be treated as a no.",
                            ))
```

Reply:
Yeah I agree that the source code looks better without the `\n` in it but I don't like how it squishes all the response information at the end of the screen. 

![image](https://user-images.githubusercontent.com/33477372/206804721-56ee8c07-c1d8-41a8-a386-f8ec47076a3c.png)
To me this just feels like a textblock that is a little too long. 

![image](https://user-images.githubusercontent.com/33477372/206804817-62e594ea-9c31-4801-8762-e473a3dcbef0.png)
I feel like this is a little easier to read since it's quite clear what you need to do. (Also ignore the missing space between the sentences, that has been fixed) 

From the perspective of the user experience I'd say that the new lines make it more clear and readable.

Reply:
Ah what I mean is, we have the `\n` outside the strings. You can see in my suggested code that I do `"\n".join()` instead of `"str1\nstr2"`.
That and also I think the line that calls for user's action should be the last line.

Reply:
using `"\n".join()` has been worked into the new update to move  "Type `yes` to confirm" onto a new line.